### PR TITLE
Prepare 0.42.0 release

### PR DESCRIPTION
**Pull request by nfagerlund**
## Description

Hello! It's time again for a release of the tfe provider. Here's the changelog update and the version number bumps. 

## Output from acceptance tests

As you can see in the Actions tab, the nightly TFE tests are currently **passing.**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
-## Unreleased
+## v0.42.0 (January 31, 2023)
```
Comment (annawinkler) [inline CHANGELOG.md:1]: A nice number! ✨

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 
-BUG FIXES:
-* `r/tfe_workspace`: Return all workspace safe deletion errors by @skeggse ([#758](https://github.com/hashicorp/terraform-provider-tfe/pull/758))
+FEATURES:
+* **New Provider Config**: `organization` (or the `TFE_ORGANIZATION` environment variable) defines a default organization for all resources, making all resource-specific organization arguments optional. ([#762](https://github.com/hashicorp/terraform-provider-tfe/pull/762))
```
Comment (annawinkler) [inline CHANGELOG.md:4]: I also DMd you about this - I like it when we list the person whose PR is included, like below:

```suggestion
* **New Provider Config**: `organization` (or the `TFE_ORGANIZATION` environment variable) defines a default organization for all resources, making all resource-specific organization arguments optional. @brandonc ([#762](https://github.com/hashicorp/terraform-provider-tfe/pull/762))
```

What are your thoughts?

**Comment by nfagerlund** [comment]
@annawinkler added! Thanks for catching that, I meant to and spaced.

**Review by annawinkler — APPROVED** [review]
🥇 Looks great!

**Pull request merged**